depthai_ros_driver/include/depthai_ros_driver/img_detections_stamper.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+/**
+ * @file img_detection_stamper.hpp
+ * @brief Publish an image detection (NN) with the timestamp from the latest Image from sensor
+ */
+#include <ros/ros.h>
+#include <depthai_datatype_msgs/ImgDetectionsStamped.h>
+#include <depthai_datatype_msgs/RawImgDetections.h>
+#include <sensor_msgs/Image.h>
+
+namespace rr {
+template <class Node>
+struct ImgDetectionsStamper : Node {
+public:
+    ImgDetectionsStamper() = default;
+    ~ImgDetectionsStamper() = default;
+
+private:
+    void onInit() override;
+    void __img_callback__(const sensor_msgs::ImageConstPtr& img_msg);
+    void __nn_callback__(const depthai_datatype_msgs::RawImgDetectionsConstPtr& nn_msg_ptr);
+    bool is_header_received_ = false;
+    std_msgs::Header last_recv_header_;
```

Comment:
Thoughts on storing the shared_ptr of the img msg instead of copy-ing the header?

Reply:
sure, using ptr in the last commit